### PR TITLE
Add ccx-upgrades-data-eng caching scenarios

### PR DESCRIPTION
**Pull request by epapbak**
# Description

Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Fixes # (issue)

## Type of change

Please delete options that are not relevant.

- Bug fix (non-breaking change which fixes an issue)
- New test feature file
- New test scenario added into existing feature file
- Non-breaking change in test steps implementation
- Breaking change (fix or feature that would cause existing functionality to not work as expected)
- This change requires a documentation update
- Refactor (refactoring code, removing useless files)
- Bump-up dependent library (no changes in the code)
- Security fix in dependent library (no changes in the code)
- Security fix in dependent library (changes made in the code)
- Documentation update
- Configuration update

## Testing steps

Please describe how the change was tested locally. If, for some reason, the testing was not done or not done fully, please describe what are the testing steps.

## Checklist
* [ ] Pylint passes for Python sources
* [ ] sources has been pre-processed by Black
* [ ] updated documentation wherever necessary
* [ ] new tests can be executed both locally and within docker container

### CODE REVIEW

File: features/ccx-upgrades-data-eng/caching.feature [hunk 1]
```diff
@@ -0,0 +1,220 @@
...
+    And The response is identical to the previous response for aaaaaaaa-bbbb-cccc-dddd-000000000000
+    When I request the cluster endpoint in localhost:8000 with path 44444444-3333-2222-1111-111111111111/upgrade-risks-prediction
+    Then The status code of the response is 200
+    And The response is identical to the previous response for 44444444-3333-2222-1111-111111111111
+    When I request the cluster endpoint in localhost:8000 with path 00000000-1111-2222-3333-444444444444/upgrade-risks-prediction
+    Then The status code of the response is 404
+    And The response is identical to the previous response for 00000000-1111-2222-3333-444444444444
+    # Item for aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee is not in the cache anymore, so this fails as RHOBS / inference are not available
+    When I request the cluster endpoint in localhost:8000 with path aaaaaaaa-bbbb-cccc-dddd-eeeeeeeeeeee/upgrade-risks-prediction
+    Then The status code of the response is 500
```
Comment (juandspy) [inline features/ccx-upgrades-data-eng/caching.feature:141]: it should be 503 IIRC

Comment (juandspy) [reply features/ccx-upgrades-data-eng/caching.feature:141]: or maybe the 500 is mapped to 503 in smart proxy

**Review by juandspy — APPROVED** [review]

File: features/ccx-upgrades-data-eng/caching.feature (hunk 1 above)
Comment (epapbak) [reply features/ccx-upgrades-data-eng/caching.feature:141]: probably. In the upgrades-end service itself, it's 500

File: features/ccx-upgrades-data-eng/caching.feature [hunk 2]
```diff
@@ -0,0 +1,219 @@
...
+  # You have to add 127.0.0.1   localhost mock-oauth2-server
+  # to your /etc/hosts. This is needed to run these tests in Gitlab CI.
+
+  Background: Data eng service is running and well configured to work
+    Given The mock CCX Inference Service is running on port 9090
+    And The mock RHOBS Service is running on port 9091
+
+
+  Scenario: Check that nothing is cached if CACHE_ENABLED is set to 0
+    Given The CCX Data Engineering Service is running on port 8000 with envs
```
Comment (tisnik) [inline features/ccx-upgrades-data-eng/caching.feature:12]: I'd probably move it to background

File: features/steps/ccx_data_engineering_service.py [hunk 3]
```diff
@@ -45,3 +45,26 @@ def start_ccx_upgrades_data_eng(context, port):
...
+
+@given("The mock RHOBS Service is running on port {port:d}")
+def start_RHOBS_mock_service(context, port):
+    """Run RHOBS service mock for a test and prepare its stop."""
+    params = ["uvicorn", "mocks.rhobs.rhobs_service:app", "--port", str(port)]
+
+    f = open(f"logs/ccx-upgrades-data-eng/{context.scenario}.log", "w")
+    popen = subprocess.Popen(params, stdout=f, stderr=f)
+    assert popen is not None
+    time.sleep(0.5)
```
Comment (tisnik) [inline features/steps/ccx_data_engineering_service.py:58]: prob. better to active check for port to be accessible?

File: features/steps/ccx_inference_service.py [hunk 4]
```diff
@@ -31,3 +32,26 @@ def start_ccx_inference_service(context, port):
...
+
+@given("The mock CCX Inference Service is running on port {port:d}")
+def start_ccx_inference_mock_service(context, port):
+    """Run ccx-inference-service for a test and prepare its stop."""
+    params = ["uvicorn", "mocks.inference-service.inference_service:app", "--port", str(port)]
+
+    f = open(f"logs/ccx-upgrades-inference/{context.scenario}.log", "w")
+    popen = subprocess.Popen(params, stdout=f, stderr=f)
+    assert popen is not None
+    time.sleep(0.5)
```
Comment (tisnik) [inline features/steps/ccx_inference_service.py:45]: dtto

Comment (tisnik) [reply features/steps/ccx_inference_service.py:45]: + prob. worth to refactor, as it is the same code as above

**Review by tisnik — APPROVED** [review]
ok, just left comments

File: features/ccx-upgrades-data-eng/caching.feature (hunk 2 above)
Comment (epapbak) [reply features/ccx-upgrades-data-eng/caching.feature:12]: The env vars change so I can "setup" the cache on each test.

File: features/steps/ccx_data_engineering_service.py (hunk 3 above)
Comment (epapbak) [reply features/steps/ccx_data_engineering_service.py:58]: true, let me change that

**Review by Bee-lee — APPROVED** [review]
LGTM overall

File: features/steps/ccx_inference_service.py (hunk 4 above)
Comment (epapbak) [reply features/steps/ccx_inference_service.py:45]: definitely. I'll merge this PR as-is so I can verify  https://gitlab.cee.redhat.com/ccx/ccx-upgrades-data-eng/-/merge_requests/43/ against the BDDs. But I'm already working on the changes you mentioned.

**Pull request merged**